codegen/src/generator/mod.rs

```diff
@@ -262,11 +287,8 @@ fn generate_struct_fields<P>(service: &Service, shape: &Shape, shape_name: &str,
 
 impl Operation {
     pub fn error_type_name(&self) -> String {
-        match &self.name[..] {
-            // EC2 has a different struct type called CancelSpotFleetRequestsError
-            "CancelSpotFleetRequests" => "CancelSpotFleetRequestsErrorType".to_owned(),
-            _ => format!("{}Error", self.name)
-        }
+        let type_name = mutate_type_name(&self.name);
+        format!("{}Error", type_name)
```

Comment:
The latter.  The only place `S3Error` is actually used at all is to define `pub type Errors = Vec<S3Error>;`, and that's the type that gets used everywhere `error_type_name()` is called, so it doesn't end up duplicated as `S3ErrorError`